### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by kelvin2go**

### CODE REVIEW

File: lib/airtable.ts [hunk 1]
```diff
@@ -0,0 +1,145 @@
+
+import Airtable from 'airtable';
+require('dotenv').config();
+
+const base = new Airtable({ apiKey: process.env.AIRTABLE_API_KEY }).base('appUW06bs08YxzVDM')
+
+const getIdea = async ({ id }) => {
+  try {
+    // 1. Find Idea by by ID   
+    const record = await base('Ideas').find(id)
```
Comment (ggho) [inline lib/airtable.ts]: miss `;` at this line? @kelvin2go

File: lib/airtable.ts [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+    // 2. Fetch all Actions (with titles and type) of this idea
+    const filterStr = record.fields['Actions'].reduce((acc, recID) => {
```
Comment (ggho) [inline lib/airtable.ts]: That's how you use `await`! very neat 👍

File: lib/airtable.ts [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+    }).firstPage();
+    return {
+      record,
+      actionRecords
+    }
+  } catch (err) {
+    console.log(err);
+  }
+  return {
+    record: [] as any,
```
Comment (ggho) [inline lib/airtable.ts]: `record: {}` instead?

File: lib/airtable.ts [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+  const userRecs = await base('Users').select({
+    view: 'Grid view',
+    filterByFormula: `{User Id} = '${user.id}'`, //use user id here as username might change
+  }).firstPage();
+  return userRecs.length === 0 ?
+    await base('Users').create({
+      "Username": user.username,
+      "User Id": String(user.id),
+    }, { typecast: true })
+    : userRecs[0]
```
Comment (ggho) [inline lib/airtable.ts]: 😍 looks so neat!

File: lib/airtable.ts [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+    const userRecIdArr = eachAction.fields['By Users'] || [];
+    if (userRecIdArr.includes(userRecord.id)) {
+      return eachAction
+    }
+  })
+  return eachAction
+};
+
+const addUsersOnAction = async ({ selectedActionId, userRecord, existingSupporters }) => {
+  if (existingSupporters.includes(userRecord.id)) return
```
Comment (ggho) [inline lib/airtable.ts]: Sorry if it's a silly question:
so it's more preferred to have no `;` after return nowadays?

File: lib/airtable.ts [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+  const userRecord = await getUserOrRegister({ user })
+
+  // 3. Fetch all sibling Actions 
+  const idea = await getIdea({ id: ideaId })
+
+  // 4. Clear any user's previous selection 
+  const lastSelectedAction = await getUserLastActionRec({ userRecord, actionRecords: idea.actionRecords });
+  if (lastSelectedAction) {
+    await removeUsersOnAction({ selectedActionId: lastSelectedAction.id, userRecord, existingSupporters: lastSelectedAction.fields['By Users'] })
+  }
```
Comment (ggho) [inline lib/airtable.ts]: Thank you! ❤️

File: lib/locale-zh.json [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+    "submitidea": {
+      "thankYouMsg": "嘩！好啊！\n有橋出橋💡，有力出力💪！\n💡出橋這裹：https://airtable.com/shrYwXgCML9aN2dI3",
+    }
+  },
+  "greeting": "歡迎你 come 幫！",
+  "unknown": "Sorry, 我唔係好明。",
+  "intro": "你想做咩？\n\n/browseideas - 睇今期 Top Ideas！\n/submitidea - 有 Idea? 出橋啦\n",
+  "helpMsg": "CommBond - Community Bonding.\n一個社區實驗平台，連結社區持份者、同路人共創共享互助生活社區。\n\n你可以透過 CommBond bot /browseideas 瀏覽平台上民間發起的社區 Ideas，然後作出回應或支持。你也可以 /submitidea 出橋上 CommBond 平台，在社區發起實驗，召集同路人用行動支持你的 Idea。\n\n",
+  "contact": "有其他疑問或意見？歡迎聯絡我們的人類：@CommBond\n"
+}
```
Comment (ggho) [inline lib/locale-zh.json]: 👍

File: lib/airtable.ts [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+    await unselectAction({ lastSelectedActionRec: lastSelectedAction, userRecord });
+  }
+  // 5. Update Actions with user's newly selected Action
+  const updatedRecord = await addUsersOnAction({ selectedActionId, userRecord, existingSupporters })
+  return {
+    updatedIdeaRecord: {
+      ...idea.record,
+      fields: {
+        ...idea.record.fields,
+        'Participation Count': updatedRecord &&  updatedRecord.hasOwnProptery('fields') && updatedRecord.fields['Action Type'] === 'Participate' ? idea.record.fields['Participation Count'] + 1 : idea.record.fields['Participation Count'],
```
Comment (ggho) [inline lib/airtable.ts]: This line is throwing this error, how to handle?
`  TypeError: updatedRecord.hasOwnProptery is not a function`

Comment (ggho) [reply lib/airtable.ts]: @kelvin2go  The counts actually don't show properly because of this error I guess.

File: lib/airtable.ts (hunk 1 above)
Comment (kelvin2go) [reply lib/airtable.ts]: es6 allow no ; 
normally  set a standard all no or all have. 

lets set all have

File: lib/airtable.ts (hunk 3 above)
Comment (kelvin2go) [reply lib/airtable.ts]: ok

**Pull request merged**